### ISSUE TITLE
No error when checking for INP001, but `--add-noqa` adds directive

### ISSUE THREAD

**Issue by hofrob** [charliermarsh/ruff#2830]
Python package:

```
foo/__init__.py
foo/bar.py  # empty
```

Contents of `__init__.py`:
```py
from foo import bar
```

Executing `ruff check -n --isolated --select INP001 foo/` does not show any error messages. But `ruff check -n --isolated --select INP001 --add-noqa foo/` adds a noqa directive.

Contents of `__init__.py` after `--add-noqa`:
```py
from foo import bar  # noqa: INP001
```

I don't think there's anything wrong with the code, so `--add-noqa` should not add any directives (in my opinion).

```
$ ruff --version
ruff 0.0.245
$ python --version
Python 3.11.2
```

:football:

**Comment by charliermarsh** [comment]
Oh very weird!

**Comment by charliermarsh** [comment]
Ahhh right, there was a TODO to thread through the package here. Will fix...
